### PR TITLE
GitHub Actions (#1)

### PR DESCRIPTION
**Pull request by jmaanmies**
* GitHub Actions

Fix dependencies for a successful make and run it

Intended as a starting point for https://github.com/latchset/mod_auth_mellon/issues/5

### CODE REVIEW

**Comment by jhrozek** [comment]
Awesome, this is much better than no CI we have now :)
Do you know what CFLAGS are used? We might want to use something strict(-ish) to make sure we don't build code by accident..

**Comment by jmaanmies** [comment]
My naive assumption is that it would use the CFLAGS defined by the project Makefile, but if we want to we can easily add environment variables to the Actions.

This seems to be a good starting point to read about stuff:
https://help.github.com/en/actions/automating-your-workflow-with-github-actions

This covers ENV variables more specifically:
https://help.github.com/en/actions/automating-your-workflow-with-github-actions/using-environment-variables

And here is a random workflow I found that uses env-vars, and in this case uses cmake and is a lot more advanced in general, showing some of the powers of the tool

I just put in the minimum work needed to get it to build, hoping to find other people to cooperate with to make it better!

**Comment by jmaanmies** [comment]
Forgot to paste link: https://github.com/KomodoPlatform/antara-gaming-sfml-template/blob/master/.github/workflows/build.yml

**Comment by jmaanmies** [comment]
You can inspect the results of a build here:  https://github.com/jmaanmies/mod_auth_mellon/commit/5f8017721419076f7ffd6354be95fe9fb7e331fa/checks?check_suite_id=344846078

**Comment by jhrozek** [comment]
I'm sorry about the delay. I think we should just merge the patch and iterate on it. I would specifically like to add builds in a centos container, but there's no point in delaying this PR, it would already be helpful on its own.

Thank you for the contribution!

File: .github/workflows/ccpp.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+name: C/C++ CI
+
+on: [push]
```
Comment (jhrozek) [inline .github/workflows/ccpp.yml]: Actually, would you mind changing on push to `on: [push, pull_request]` ?

Comment (jmaanmies) [reply .github/workflows/ccpp.yml]: done!

**Comment by jmaanmies** [comment]
> Sorry I ask for changes after letting the PR sit for 11 days, but can you just change the on from push to push, pull_request?

Hey! Just got time to look at this after some traveling. Fixed the requested change!

**Comment by jhrozek** [comment]
So, I'm going to squash the patches locally and push to master.

**Comment by jhrozek** [comment]
Pushed in 919bce825da3e6b85ae82b7f02fab68cf47f927e

**Pull request closed without merging**